phpBB/phpbb/install/converter/module/converter/task/convert.php

```diff
@@ -0,0 +1,145 @@
+	 * @param \phpbb\install\helper\config                        $install_config
+	 * @param \phpbb\install\helper\iohandler\iohandler_interface $iohandler
+	 * @param \phpbb\install\helper\container_factory             $container_factory
+	 * @param \phpbb\language\language                            $language
+	 * @param                                                     $phpbb_root_path
+	 */
+	public function __construct(\phpbb\install\converter\bin\converter $converter, $helper,
+		\phpbb\install\helper\config $install_config,
+		\phpbb\install\helper\iohandler\iohandler_interface $iohandler,
+		\phpbb\install\helper\container_factory $container_factory,
+		\phpbb\language\language $language, $phpbb_root_path)
+	{
+		$this->helper = $helper;
+		$this->install_config = $install_config;
+		$this->io_handler = $iohandler;
+		$this->converter = $converter;
+		$this->phpbb_root_path = $phpbb_root_path;
+		$this->container_factory = $container_factory;
+		$this->language = $language;
+		parent::__construct(true);
+	}
+	/**
+	 * {@inheritdoc}
+	 */
+	public function run()
+	{
+		try
+		{
+			$this->helper->set_conversion_status(true);
+			$yaml_queue = $this->helper->get_yaml_queue();
+			$curr_index = $this->helper->get_file_index();
+			if ($this->helper->get_conversion_status() && $curr_index < count($yaml_queue))
+			{
+				if (!$this->helper->get_chunk_status() || $this->helper->get_chunk_status() === null)
+				{
+					$this->helper->set_current_conversion_file($yaml_queue[$curr_index]);
+					$this->io_handler->add_log_message('Loading..', 'Fetching next file');
+					$this->helper->set_current_chunk(0);
+					$this->helper->set_chunk_status(true);
+					$log_msg = $this->language->lang('CF_LOG_CONVERTING') . $yaml_queue[$curr_index];
+					$this->io_handler->set_task_count(1, true);
+					$this->io_handler->set_progress($log_msg, 0.01); //Gives 1 % value at progress bar initially
+					$this->io_handler->add_log_message($this->language->lang('CF_LOG_CONVERTING'), $log_msg);
+					$this->io_handler->send_response();
+				}
+				else
+				{
+					$total_chunks = $this->helper->get_total_chunks();
+					$chunk = $this->helper->get_current_chunk();
+					$log_msg = $this->language->lang('CF_LOG_CONVERTING') . $yaml_queue[$curr_index] . " " . $this->language->lang('CF_LOG_PART') . "[ " . ($chunk + 1) . " ]";
+					$this->io_handler->add_log_message($this->language->lang('CF_LOG_CONVERTING'), $log_msg);
+					$this->io_handler->set_task_count($total_chunks);
+					$this->io_handler->set_progress($log_msg, ($chunk + 1));
+					$this->helper->set_current_chunk($chunk + 1);
+					$this->io_handler->send_response();
+				}
+				$this->converter->begin_conversion($yaml_queue[$curr_index], $this->helper, $this->io_handler);
+				if (!$this->helper->get_chunk_status())
+				{
+					$this->helper->next_file($curr_index);
+					sleep(2); //sleeps 2 seconds to prevent abrupt change of progress bar.
```

Comment:
Its a fix kind of to make those transitions look smooth. Else it shows a sudden jump.